x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java

```diff
@@ -210,6 +214,7 @@ private void notifyListeners(String jobId, @Nullable AcknowledgedResponse ack, @
     private void normalDeleteJob(ParentTaskAssigningClient parentTaskClient, DeleteJobAction.Request request,
                                  ActionListener<AcknowledgedResponse> listener) {
         String jobId = request.getJobId();
+        memoryTracker.removeJob(jobId);
```

Comment:
The problem is that the refresh doesn't remove entries for job IDs that don't have persistent tasks.  The reason for that is that there's a race between opening and periodic refreshing - a periodic refresh could be part way through when an open occurs, and if the periodic refresh removed entries that didn't have persistent tasks at the start of the refresh then it would undo the work of the open.

I'll add a comment to explain this.

Reply:
I improved the comments in 10a1467680452a30dcb4f9403e8d8931b0fa4bf2